core/src/main/scala/org/http4s/internal/parsing/Rfc3986.scala

```diff
@@ -9,14 +9,27 @@ package org.http4s.internal.parsing
 import cats.parse.Parser.{`end`, char, charIn, string1}
 import cats.parse.{Parser, Parser1}
 import cats.parse.Rfc5234.{digit, hexdig}
-import org.http4s.Uri.Ipv6Address
+import org.http4s.Uri.{Ipv4Address, Ipv6Address}
 
 /** Common rules defined in Rfc3986
   *
   * @see [[https://tools.ietf.org/html/rfc3986]]
   */
 private[http4s] object Rfc3986 {
 
+  val ipv4: Parser1[Ipv4Address] = {
+    val decOctet = (char('1') ~ digit ~ digit).backtrack
+      .orElse1(char('2') ~ charIn('0' to '4') ~ digit).backtrack
+      .orElse1(string1("25") ~ charIn('0' to '5')).backtrack
+      .orElse1(charIn('1' to '9') ~ digit).backtrack
+      .orElse1(digit).string.map(_.toInt.toByte).backtrack
+
+    val dot = char('.')
+    (decOctet ~ dot ~ decOctet ~ dot ~ decOctet ~ dot ~ decOctet).map {
```

Comment:
This is another one that can use `<*` or `*>` around the dots.  Then your case is only four parens instead of seven.

GitHub won't let me respond to your other question about getting a `Short` instead of a `(Short, Short)`.  I think it's an operator precedence issue.  I bet this would work:

`(decOctet ~ (dot *> decOctet) ~ (dot *> decOctet) ~ (dot *> decOctet))`

Reply:
Ah, that makes sense! Didn't think about the precedence issue. I will update that where it is beneficial. Thanks!